src/hankel.jl

```diff
@@ -0,0 +1,116 @@
+# Hankel
+struct Hankel{T<:Number} <: AbstractMatrix{T}
+    v::AbstractVector{T}
+    s::Dims{2} # size
+end
+
+Hankel{T}(v::AbstractVector, h::Integer, w::Integer) where T = Hankel{T}(v,(h,w))
+Hankel(v::AbstractVector, h::Integer, w::Integer) = Hankel{eltype(v)}(v,h,w)
+Hankel(v::AbstractVector) = Hankel(v,((l+1)÷2,(l+1)÷2)) # square by default
+function Hankel(vc::AbstractVector, vr::AbstractVector)
+    if vc[end] != vr[1]
+        throw(ArgumentError("vc[end] != vr[1]"))
+    end
+    Hankel(vcat(vc,vr[2:end]), (length(vc), length(vr)))
+end
+
+function getproperty(A::Hankel, s::Symbol)
+    if s==:vc
+        A.v[1:A.s[1]]
+    elseif s==:vr
+        A.v[A.s[1]:end]
+    else
+        getfield(A,s)
+    end
+end
+
+# convert from general matrix to Hankel matrix
+Hankel{T}(A::AbstractMatrix, uplo::Symbol = :L) where T<:Number = convert(Hankel{T}, Hankel(A,uplo))
+# using the first column and last row
+function Hankel(A::AbstractMatrix, uplo::Symbol = :L)
+    s=size(A)
+    if uplo == :L
+        if s[1] == ℵ₀
```

Comment:
Don't do this. Instead do:

```julia
Hankel(A::AbstractMatrix, uplo::Symbol = :L) = Hankel(A::AbstractMatrix, uplo, size(A))
function Hankel(A::AbstractMatrix, uplo::Symbol = :L, _)
    if uplo == :L
         Hankel(A[:,1],A[end,:])
   elseif ...
      ...
   end
end
```
Then `Hanke(A, uplo, ::Tuple{InfiniteCardinal{0},Any}})` can be overloaded in another package (e.g. InfiniteLinearAlgebra.jl)

Reply:
What's the purpose of the third argument?

Reply:
So that it can be overloaded in the infinite case in another package

Reply:
Actually, here's another idea: just use `isfinite` to test if the size is finite here.